src/main/java/org/threadly/util/debug/Profiler.java

```diff
@@ -231,6 +311,30 @@ public void run() {
             throw ExceptionUtils.makeRuntime(e.getCause());
           }
         }
+        // start or schedule to handle run time limit
+        if (sampleDurationInMillis > 0) {
+          if (executor instanceof SimpleSchedulerInterface) {
```

Comment:
The whole providing an executor is awkward...here if it is a scheduler (yucky ugly instanceof check I know) we schedule a task to call stop after the duration.  If an executor is null, or just a simple executor we create a new thread.  I did not think it would be worth taking a second thread on the executor to just block in a Thread.sleep() till we are ready to stop.

I could not let the profiling thread stop itself because it can't hold any references to this parent class which needs to have stop called.  Otherwise this class will never have the possibility of being garbage collected.  As implemented right now if you passed this in WITHOUT a timeout and then let the profiler reference go null, the profiler would automatically stop itself.  If you passed it in WITH a timeout, the profiler would continue to exist until this timeout thread ran, stopped it, then allowed for GC.

Let me know if you think we can make this cleaner in any way.  Honestly since these threads are pretty long running I wish we never accepted an Executor, and instead always just created threads.